src/Infrastructure/Mesh/src/Legacy/ESMCI_MeshMerge.C

```diff
@@ -1178,12 +1178,12 @@ void concat_meshes(const Mesh & srcmesh, const Mesh & dstmesh, Mesh & mergemesh,
                       for(int npt=0; npt<clip_num_nodes; npt++) std::cout << clip_cd_sph[npt*2] << "  " << clip_cd_sph[npt*2+1] << std::endl; 
                       delete[] clip_cd_sph; 
                     }
-                    char msg[1024];
-                    sprintf(msg," - there was a problem (e.g. repeated points, clockwise poly, etc.) with the triangulation of the element:\n"); 
-                    sprintf(msg,"%selem Id: %d clip_elem Id: %d\n", msg, elem.get_id(), clip_elem.get_id());
+                    std::stringstream msg;
+                    msg << " - there was a problem (e.g. repeated points, clockwise poly, etc.) with the triangulation of the element:\n";
+                    msg << "elem Id: " << elem.get_id() << " clip_elem Id: " << clip_elem.get_id() << "\n";
                     {
-                      cd_sph = new double[num_p*2];   cart2sph(num_p, pts, cd_sph); 
-                      for(int npt=0; npt<num_p*2; npt++) sprintf(msg,"%s %g", msg, cd_sph[npt]); 
+                      cd_sph = new double[num_p*2];   cart2sph(num_p, pts, cd_sph);
+                      for(int npt=0; npt<num_p*2; npt++) msg << " " << cd_sph[npt];
```

Comment:
Note that this change is untested because it is inside an `#if 0` block.

I tested it by:

(1) Testing the build when I changed `#if 0` to `#if 1` in the surrounding block

(2) Writing the following test program that checks the msg string before and after this change, ensuring that the resulting string is identical before and after:

```C++
#include <iostream>
#include <sstream>

using namespace std;

int main() {
   int num_p = 4;
   double *cd_sph = new double[num_p * 2];
   for (int npt = 0; npt < num_p * 2; npt++) {
      cd_sph[npt] = 24 + 36.003*npt;
   }

   char msg[1024];
   sprintf(msg, " - there was a problem (e.g. repeated points, clockwise poly, etc.) with the triangulation of the element:\n");
   sprintf(msg, "%selem Id: %d clip_elem Id: %d\n", msg, 1005, 123456);
   for (int npt = 0; npt < num_p * 2; npt++)
      sprintf(msg, "%s %g", msg, cd_sph[npt]);

   std::stringstream msg2;
   msg2 << " - there was a problem (e.g. repeated points, clockwise poly, etc.) with the triangulation of the element:\n";
   msg2 << "elem Id: " << 1005 << " clip_elem Id: " << 123456 << "\n";
   for (int npt = 0; npt < num_p * 2; npt++)
      msg2 << " " << cd_sph[npt];

   cout << msg << "\n";
   cout << msg2.str() << "\n";
}
```